### PR TITLE
add ci

### PR DESCRIPTION
**Pull request by Gsantomaggio**
closes https://github.com/qweeze/rstream/issues/13

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Run image
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: ${{ matrix.poetry-version }}
+      - name: poetry install
+        run: poetry install
+      - name: poetry run pytest
+        run: poetry run pytest
```
Comment (qweeze) [inline .github/workflows/test.yaml:33]: Never used actions before, but if I understand correctly rabbit's host will be mapped to the service name - `rabbitmq-streaming` so I guess we need to invoke pytest with `pytest --rmq-host rabbitmq-streaming`

Is there a way to run this before merging? Or perhaps we can debug this in a next PR

**Review by qweeze — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (Gsantomaggio) [reply .github/workflows/test.yaml:33]: It uses `localhost:5552`.  We use it in all [the clients](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/main/.github/workflows/test_and_publish_image.yml#L8-L24)

**Comment by qweeze** [comment]
I also use some linting/formatting tools locally in a pre-commit hook:
```bash
python -m pytest
isort --check-only .
black --check .
flake8 --exclude=venv,local_tests --max-line-length=120 --ignore=E203
mypy .
```
I think we probably should add them to CI as well, to maintain consistent code style

Also, currently I publish package releases to pypi manually, I guess it can be done with an action too. Added an issue for that - #20

**Comment by Gsantomaggio** [comment]
>I also use some linting/formatting tools locally in a pre-commit hook:




added via https://github.com/qweeze/rstream/pull/19/commits/1405a17a9b23716874da5a28847a5ce1ba35865a

**Comment by Gsantomaggio** [comment]
@qweeze I would merge! If there are problems we can fix them later. At least we see if it works!

**Pull request merged**